src/backend/columnar/columnar_customscan.c

```diff
@@ -821,12 +863,50 @@ static void
 ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 							   ExplainState *es)
 {
-	TableScanDesc scanDesc = node->ss.ss_currentScanDesc;
+	ColumnarScanDesc columnarScanDesc = (ColumnarScanDesc) node->ss.ss_currentScanDesc;
 
-	if (scanDesc != NULL)
+	if (columnarScanDesc != NULL)
 	{
-		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
+		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(columnarScanDesc);
 		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
 							   chunkGroupsFiltered, es);
 	}
+
+	char *projectedColumnsStr = ColumnarScanProjectedColumnsStr(node, ancestors, es);
+	if (projectedColumnsStr)
+	{
+		ExplainPropertyText("Columnar Projected Columns", projectedColumnsStr, es);
+	}
+}
+
+
+/*
+ * ColumnarScanProjectedColumnsStr generates projected column string for
+ * explain output. Returns NULL if no columns will be projected.
+ */
+static char *
+ColumnarScanProjectedColumnsStr(CustomScanState *node, List *ancestors, ExplainState *es)
+{
+	ScanState *scanState = &node->ss;
+	List *neededVarList = ColumnarVarNeeded(scanState);
+	int natts = RelationGetNumberOfAttributes(scanState->ss_currentRelation);
+	List *uniqueVarList = DeduplicateVarList(neededVarList, natts);
+	if (list_length(uniqueVarList) == 0)
+	{
+		return NULL;
+	}
```

Comment:
Maybe return something like `<no columns will be read>` so that users can understand columnar optimized out all the columns.